src/main/scala/firrtl/passes/RemoveAccesses.scala

```diff
@@ -66,17 +65,27 @@ object RemoveAccesses extends Pass {
         (l, i) <- ls.zipWithIndex
         if ((i % stride) >= start) & ((i % stride) < end)
       ) yield l
-    case e: WSubAccess =>
-      val ls = getLocations(e.expr)
-      val stride = get_size(e.tpe)
-      val wrap = e.expr.tpe.asInstanceOf[VectorType].size
-      ls.zipWithIndex.map {
-        case (l, i) =>
-          val c = (i / stride) % wrap
-          val basex = l.base
-          val guardx = AND(l.guard, EQV(UIntLiteral(c), e.index))
-          Location(basex, guardx)
+    case SubAccess(expr, index, tpe, _) =>
+      getLocations(expr).zipWithIndex.flatMap {
+        case (Location(exprBase, exprGuard), exprIndex) =>
+          getLocations(index).map {
+            case Location(indexBase, indexGuard) =>
+              Location(
+                exprBase,
+                AND(
+                  AND(
+                    indexGuard,
+                    exprGuard
+                  ),
+                  EQV(
```

Comment:
I think this expression could be cached, but IMO this is such a rare occurrence I don't think its too bad.